### PR TITLE
nixos/fprot: add type

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change
#76184

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/security/fprot.nix [hunk 1]
```diff
@@ -16,16 +16,19 @@ in {
           description = ''
             product.data file. Defaults to the one supplied with installation package.
           '';
+          type = types.path;
```
Comment (cole-h) [inline nixos/modules/services/security/fprot.nix:19]:
```suggestion
          type = types.either types.path types.str;
```

File: nixos/modules/services/security/fprot.nix [hunk 2]
```diff
@@ -16,16 +16,19 @@ in {
...
         frequency = mkOption {
           default = 30;
+          type = types.int;
           description = ''
             Update virus definitions every X minutes.
           '';
         };
 
         licenseKeyfile = mkOption {
+          type = types.path;
```
Comment (cole-h) [inline nixos/modules/services/security/fprot.nix:31]:
```suggestion
          type = types.either types.path types.str;
```

**Review by cole-h — CHANGES_REQUESTED** [review]

File: nixos/modules/services/security/fprot.nix (hunk 1 above)
Comment (Infinisil) [reply nixos/modules/services/security/fprot.nix]: I was asked by @cole-h whether this makes sense to allow both paths and derivations. However this is in fact not necessary, `types.path` works for both paths and derivations already, so I think this change can be reverted, same for `licenseKeyfile` :)

Comment (cole-h) [reply nixos/modules/services/security/fprot.nix]: Sorry, @Scriptkiddi :grimacing:

**Review by cole-h — APPROVED** [review]
Diff LGTM, thanks!

**Pull request merged**